src/vs/workbench/contrib/positronOutputWebview/browser/downloadUtils.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export type PositronDownloadMessage = {
+	type: 'positronDownload';
+	data: string | ArrayBuffer | null;
+	downloadName: string;
+};
+
+export function msgIsDownloadMessage(msg: any): msg is PositronDownloadMessage {
+	return msg.type === 'positronDownload';
+}
+
+// Let typescript know that the vscode object is available
+declare const vscode: {
+	postMessage(message: PositronDownloadMessage): void;
+};
+
+// Function is meant to be dependency free so it can be serialized into the webview with the
+// Function.toString() method
+function handleWebviewClicks() {
+
+	// eslint-disable-next-line no-restricted-syntax
+	document.addEventListener('click', (event) => {
+		const suppressEvent = () => {
+			event.preventDefault();
+			event.stopPropagation();
+		};
+		for (const node of event.composedPath()) {
+			// eslint-disable-next-line no-restricted-syntax
+			if (node instanceof HTMLAnchorElement && node.href) {
+				if (node.href.startsWith('blob:')) {
+					handleBlobUrlClick(node.href, node.download);
+					suppressEvent();
+				} else if (node.href.startsWith('data:')) {
+					handleDataUrl(node.href, node.download);
+					suppressEvent();
+				} else {
+					console.log('handleDataUrl called with unknown href type', node.href);
+				}
+				break;
+			}
+		}
+	});
+
+	const handleBlobUrlClick = async (url: string, downloadName: string) => {
+		try {
+			const response = await fetch(url);
+			const blob = await response.blob();
+			const reader = new FileReader();
+			reader.addEventListener('load', () => {
+				handleDataUrl(reader.result, downloadName);
+			});
+			reader.readAsDataURL(blob);
+		} catch (e) {
+			console.error(e.message);
+		}
+	};
+
+	const handleDataUrl = async (data: string | ArrayBuffer | null, downloadName: string) => {
+		vscode.postMessage({
+			type: 'positronDownload',
+			data,
+			downloadName
+		});
+	};
+
+	// Override the prompt function to return the default value or 'myFile' if one isnt provided.
+	// This is needed because the prompt function is not supported in webviews and the prompt function
+	// is commonly used by libraries like bokeh to provide names for files to save. The main file save
+	// dialog that positron shows will already provide the ability to change the file name so we're
+	// just providing a default value here.
+	window.prompt = (message, _default) => {
+		return _default ?? 'myFile';
+	};
```

Comment:
Do we need both this and the one in `webview-events.js`?

Reply:
Also as mentioned in that one, maybe `Untitled` is more in line with VSCode than `myFile`?